bin/zopen-build

```diff
@@ -55,72 +55,96 @@ printEnvVar()
+  ZOPEN_LDFLAGS        C/C++ linker flags (default set by dependency)
+
+Restricted Usage (phpport)
+  ZOPEN_GIT_SETUP      Specify whether git files should be added to a local repo or if this will be done manually (defaults to Y)
+
+Restricted Usage (tclport)
+  ZOPEN_SRC_DIR        specify a relative source directory to cd to for bootstrap, configure, build, check, install (defaults to '.')
+
+Restricted Usage (meta)
   ZOPEN_IMAGE_REGISTRY Docker image registry to an OCI image to (use with --oci option)
   ZOPEN_IMAGE_DOCKERFILE_NAME Dockerfile name (default: Dockerfile)
   ZOPEN_IMAGE_DOCKER_NAME Docker/podman tool name (default: podman)
   ZOPEN_IMAGE_REGISTRY_ID The ID to authenticate to the Docker image registry (use with --oci option)
   ZOPEN_IMAGE_REGISTRY_KEY_FILE The file containing the key to authenticate to the Docker image registry (use with --oci option)
   ZOPEN_LOG_DIR        The directory to store build logs (defaults to '${ZOPEN_ROOT}/log')
-  ZOPEN_INSTALL        Installation program to run. If skip is specified, no installation step is performed (defaults to '${ZOPEN_INSTALLD}')
-  ZOPEN_INSTALL_OPTS   Options to pass to installation program (defaults to '${ZOPEN_INSTALL_OPTSD}')
-  ZOPEN_CLEAN          Clean up program to run (defaults to '${ZOPEN_CLEAND}')
-  ZOPEN_CLEAN_OPTS     Options to pass to clean up  program (defaults to '${ZOPEN_CLEAN_OPTSD}')
-  ZOPEN_SHELL          Specify an alternate shell to use if -s option specified (defaults to /bin/sh)"
+  ZOPEN_SHELL          Specify an alternate shell to use if -s option specified (defaults to /bin/sh)
 
+Deprecated:
+  ZOPEN_DEV_TAG        The tag that the git repo should checkout as a branch (optional)
+  ZOPEN_STABLE_TAG     The tag that the git repo should checkout as a branch (optional)
+  ZOPEN_TYPE           The type of package to download. Valid types are TARBALL, BARE and GIT 
+  ZOPEN_TARBALL_URL    The fully qualified URL that the tarball should be downloaded from (required if ZOPEN_TYPE=TARBALL)
+  ZOPEN_TARBALL_DEPS   Space-delimited set of source packages this git package depends on to build (required if ZOPEN_TYPE=TARBALL)
+  ZOPEN_GIT_URL        The fully qualified URL that the git repo should be cloned from (required if ZOPEN_TYPE=GIT)
+  ZOPEN_GIT_DEPS       Space-delimited set of source packages this tarball package depends on to build (required if ZOPEN_TYPE=GIT)
+  ZOPEN_GIT_BRANCH     The branch that the git repo should checkout 
+  ZOPEN_GIT_TAG        The tag that the git repo should checkout as a branch 
+  ZOPEN_URL            Alternate environment variable instead of ZOPEN_TARBALL_URL or ZOPEN_GIT_URL (alternate to ZOPEN_TARBALL_URL or ZOPEN_GIT_URL)
+  ZOPEN_DEPS           Alternate environment variable instead of ZOPEN_TARBALL_DEPS or ZOPEN_GIT_DEPS (alternate to ZOPEN_TARBALL_DEPS or ZOPEN_GIT_DEPS)
+  ZOPEN_DEV_TYPE       The type of package to download. Valid types are TARBALL, BARE and GIT 
```

Comment:
This is optional and in case the "heuristic" does not detect the type correctly, this can force it to the right type.

Reply:
'this' being ZOPEN_xxx_TYPE right? I moved this to deprecated because none of the repos actually use it.

Reply:
If you think it's useful but currently unused, then I can move those 2 up to that.

Reply:
Actually I think we'll need it for https://github.com/ZOSOpenTools/comp_clangport/blob/main/buildenv and the other compiler stubs once we migrate to the STABLE/DEV buildenv format or perhaps we can add logic to set it to BARE (or equivalent) if no url is specified

Reply:
moved to a new section indicating currently unused.